stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt

```diff
@@ -519,6 +529,14 @@ internal class ChatDomainImpl internal constructor(
         return currentUser.id + "-" + UUID.randomUUID().toString()
     }
 
+    private fun addTypingChannel(channelController: ChannelControllerImpl) {
+        val typingEvent: LiveData<Pair<String, List<User>>> = channelController.typing.map { userList ->
```

Comment:
Our offline support feature (`ChatDomain`) isn't used by customers without our view models. Then we could change the type of `ChannelController::typing` and include there `channelId` . Result will be `LiveData<Pair<String, List<User>>>`. It lets us to omit mapping each time when add new channelController.

Reply:
🤔  But how would we create a way to listen for typing events in all channels at the same time? I use the mapping because new channels might be added and I need to listen for updates in all active channels and, if I understand correctly, a channelController listen to only of channel.

Reply:
The same idea. When you create new `ChannelController` and add to a map of `ChatDomain` you also add its' `ChannelController::typing` as a source to all typingEvents into `ChatDomain::_typingChannels`. So implementation will be the same, we only can omit mapping of `ChannelController::typing`

Reply:
Done. I also mapped the `Pair<String, List<User>>` to `TypeEvent`